### PR TITLE
Enable PR validation for template changes

### PR DESCRIPTION
**Pull request by weikanglim**
This PR enables template tests to run as part of PR validation for modified templates.

Only the specific templates being modified are tested.

### CODE REVIEW

File: eng/scripts/Set-TemplateTestMatrixVariable.ps1 [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+$matrix = @{}
+foreach ($template in $templateNames) {
+    $jobName = $template.Replace('/', '_')
+    $matrix[$jobName] = @{ TemplateName = $template }
+}
+
+# Adding extra test for capitals letters support. Using first template
+$firstTemplate = $templateNames[0]
+$capitalsTest = $firstTemplate.Replace('/', '_') + "-Upper-case-test"
+$matrix[$capitalsTest] = @{ TemplateName = $firstTemplate; UseUpperCase = "true" }
```
Comment (benbp) [inline eng/scripts/Set-TemplateTestMatrixVariable.ps1:85]: I know you are just copying/re-organizing this code, but as I'm looking at it for the first time, I'm curious if you or @danieljurek know the reason why we need to add tests via a powershell script as opposed to just defining extras in the matrix file and using the `-Filter` or `-Replace` flags to tune what gets tested?

Comment (weikanglim) [reply eng/scripts/Set-TemplateTestMatrixVariable.ps1]: @benbp This was resolved as part of the refactoring PR, but happy to take suggestions to improve this as part of another PR in the future. The extra upper-case-test really should be redone differently -- probably as part of the CLI validation and not template test validation.

Comment (benbp) [reply eng/scripts/Set-TemplateTestMatrixVariable.ps1]: I think CLI validation makes sense as you say.

Comment (benbp) [reply eng/scripts/Set-TemplateTestMatrixVariable.ps1]: But my general approach with matrix test formulation is to try to push as much generation of tests into the matrix itself, and use the existing parameters to filter/update the matrix if needed with moderation. This way it's easier to look at a matrix config and reason about what test cases will actually run, vs. having to parse through a lot of yaml/scripts to figure out what will be tested (and it's harder to reproduce the same test runs/matrix cases locally).

**Review by vhvb1989 — APPROVED** [review]

File: templates/todo/api/java/src/main/resources/application.properties [hunk 2]
```diff
@@ -5,7 +5,9 @@ spring.jackson.serialization.WRITE_DATES_AS_TIMESTAMPS=false
 
 # Import KeyVault secrets as properties
 spring.cloud.azure.keyvault.secret.property-sources[0].enabled=true
-spring.cloud.azure.keyvault.secret.property-sources[0].endpoint=${AZURE_KEY_VAULT_ENDPOINT}
+spring.cloud.azure.keyvault.secret.property-sources[0].endpoint=${AZURE_KEY_VAULT_ENDPOINT:#{null}}
```
Comment (weikanglim) [inline templates/todo/api/java/src/main/resources/application.properties:8]: Will revert once this PR turns green

**Pull request merged**